### ISSUE TITLE
Toggle support for QuickInput in API

### ISSUE THREAD

**Issue by sudoaugustin** [microsoft/vscode#144956]
Please provide **option buttons** option for `window.createInputBox` & `window.showInputBox`.

![Group 2](https://user-images.githubusercontent.com/57541158/158027829-2822508f-8f9d-4cb4-bc81-ff8bf67102b9.png)

**Comment by TylerLeonhardt** [comment]
I like the idea. It's not something we've explored yet mostly because buttons are available in the title bar:

https://code.visualstudio.com/api/references/vscode-api#InputBox

Via the `buttons` property.

Is there a particular use case you were looking for?

**Comment by sudoaugustin** [comment]
Let's say I want to create a password input with a visibility toggle button. Displaying the eye off-on icon on the title bar seems weird UI-UX. Because the inputbox and button which tweaks the input were on different nodes & seems like there is no connection between these two (imagine putting a password-toggle icon beside the label instead of inside the inputbox ).
What if I wanna create an input box with a clear input value button?

**Comment by TylerLeonhardt** [comment]
Yeah I think that makes sense. We don't have such a concept today of toggles in the quick pick (even natively) so this is a bit more work... so I've added it to the backlog since we're not planning on addressing this right away... but we have thought about it for a while.

**Comment by TylerLeonhardt** [comment]
We do now have Toggle support for Quick Inputs in core, the next step would be lighting it up in API

**Comment by gjsjohnmurray** [comment]
Wiring toggle support through to the API would benefit this use-case:

![junk](https://github.com/user-attachments/assets/1f649b40-17cf-4520-bef4-2b92502c3880)

This quickpick is in the extension whose source is at https://github.com/intersystems-community/vscode-objectscript. The 3 titlebar buttons act as toggles. Currently the only toggle-state indication available to the user is the tooltip saying "Hide..." or "Show..."

**Comment by dmitrivMS** [comment]
Reopening for API review

**Comment by dmitrivMS** [comment]
Reopening to track for api-finalization.

**Comment by blobberim** [comment]
this has potentially broken an existing (maintenance mode) extension with vscode 1.106, see [issue](https://github.com/dendronhq/dendron/issues/4007) - as an end-user, do we have any control to change the behaviour?

**Comment by TylerLeonhardt** [comment]
This is now linked to https://github.com/microsoft/vscode/issues/221397
